task/sql-tasks.js

```diff
@@ -280,7 +429,27 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+  let result = await db.query(`
+        SELECT DISTINCT sub.CompanyName, sub.ProductName, PricePerItem
+        FROM (
+            SELECT c.CompanyName, p.ProductName, d.UnitPrice
+            FROM Customers c
+            INNER JOIN Orders o ON c.CustomerID=o.CustomerID
+            INNER JOIN OrderDetails d ON o.orderID=d.OrderID
+            INNER JOIN Products p ON p.ProductId=d.ProductID
+            ) sub
+        INNER JOIN (
+            SELECT c.CompanyName, MAX(d.UnitPrice) as PricePerItem

```

Comment:
just select from this subquery and join all information you need to this subquery


```
Select * 
From maxpricesubquery
join.....
join...
...
```